### PR TITLE
stackrox: build performance

### PR DESCRIPTION
**Pull request by gavin-stackrox**
This PR is the start of an attempt to improve the performance of `stackrox/stackrox` builds in CI. It will promote the `bin` image at merge time so that it can be used in subsequent PR and merge builds for caching. PRs and merges to the `stackrox/stackrox` `ci-perf-hacks` branch will then be used to validate a caching approach before migration to master.

This PR also create a general purpose path for experimenting with CI by allowing any `stackrox/*` `*-hacks` branches to be configured for CI.

/hold
/uncc

### CODE REVIEW

**Review by sbostick — APPROVED** [review]
LGTM

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+- dockerfile_path: Dockerfile
+  inputs:
+    src:
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/stackrox/stackrox/docs/.
+  to: docs
+- build_args:
+  - name: GO_VERSION
+    value: 1.17.2
```
Comment (porridge) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml:67]: We don't need this any more.

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  - name: ROX_IMAGE_FLAVOR
+    value: development_build
+  dockerfile_path: operator/Dockerfile
+  inputs:
+    operator-artifacts:
```
Comment (porridge) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml:72]:
```suggestion
  inputs:
    golang-1.17.2:
      as:
      - golang:1.17.2
    operator-artifacts:
```

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml [hunk 3]
```diff
@@ -0,0 +1,166 @@
+binary_build_commands: .openshift-ci/dispatch.sh binary-build-commands
+build_root:
```
Comment (porridge) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml:2]:
```suggestion
binary_build_commands: .openshift-ci/dispatch.sh binary-build-commands
base_images:
  golang-1.17.2:
    name: golang
    namespace: stackrox
    tag: 1.17.2
build_root:
```

**Review by porridge — APPROVED** [review]
/lgtm
Although this seems to be half of the picture, so I don't pretend to understand how this is supposed to work 😅

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml (hunk 3 above)
Comment (janisz) [reply ci-operator/config/stackrox/stackrox/stackrox-stackrox-ci-perf-hacks.yaml:2]: After this we should point in our golang version check script that this needs to be changed too https://github.com/stackrox/stackrox/blob/master/EXPECTED_GO_VERSION#L1

**Review by janisz — APPROVED** [review]

**Review by droslean — APPROVED** [review]

**Pull request merged**